### ISSUE TITLE
[java] - UseUnderscoresInNumericLiterals reports on `1234.123_456`

### ISSUE THREAD

**Issue by blacelle** [pmd/pmd#5987]
**Affects PMD Version:** 
7.4.0

**Rule:**

Please provide the rule name and a link to the rule documentation:
https://pmd.github.io/pmd/pmd_rules_java_codestyle.html#useunderscoresinnumericliterals

**Description:**
Relates with https://github.com/solven-eu/cleanthat/issues/896

**Code Sample demonstrating the issue:**

```java
assertEquals(1234.123456, testValue)
```

**Expected outcome:**

PMD reports a violation at line 0, but it's unclear if it is right or wrong. The requirement for `_` triggers by default on `4` digits, but it is unclear if it triggers independantly to the part before and after the `.`.

`assertEquals(1234.123456, testValue)` is OK.
`assertEquals(1_234.123_456, testValue)` is OK.
`assertEquals(1234.123_456, testValue)` is KO.

(This is reported in the context of https://github.com/solven-eu/cleanthat/issues/896, which is a rule in tool automated the cleaning of various rules (especially PMD rules), in the vein of https://github.com/pmd/pmd/issues/693).

**Comment by UncleOwen** [comment]
`1234.123_456` is a violation because once there is at least one `_` in the literal, you're supposed to use `_` consistently. That's the way the rule is written.
